cmd/zoekt-sourcegraph-indexserver/sg.go

```diff
@@ -347,7 +505,97 @@ type indexOptionsItem struct {
 	Error string
 }
 
-func (s *sourcegraphClient) getIndexOptions(fingerprint string, repos ...uint32) ([]indexOptionsItem, string, error) {
+func (o *indexOptionsItem) FromProto(x *proto.ZoektIndexOptions) {
+	branches := make([]zoekt.RepositoryBranch, 0, len(x.Branches))
+	for _, b := range x.GetBranches() {
+		branches = append(branches, zoekt.RepositoryBranch{
+			Name:    b.GetName(),
+			Version: b.GetVersion(),
+		})
+	}
+
+	item := indexOptionsItem{}
+
+	item.IndexOptions = IndexOptions{
+		RepoID:     uint32(x.GetRepoId()),
+		LargeFiles: x.GetLargeFiles(),
+		Symbols:    x.GetSymbols(),
+		Branches:   branches,
+		Name:       x.GetName(),
+
+		Priority: x.GetPriority(),
+
+		DocumentRanksVersion: x.GetDocumentRanksVersion(),
+
+		Public:   x.GetPublic(),
+		Fork:     x.GetFork(),
+		Archived: x.GetArchived(),
+	}
+
+	item.Error = x.GetError()
+
+	*o = item
+}
+
+func (o *indexOptionsItem) ToProto() *proto.ZoektIndexOptions {
+	branches := make([]*proto.ZoektRepositoryBranch, 0, len(o.Branches))
+	for _, b := range o.Branches {
+		branches = append(branches, &proto.ZoektRepositoryBranch{
+			Name:    b.Name,
+			Version: b.Version,
+		})
+	}
+
+	return &proto.ZoektIndexOptions{
+		RepoId:     int32(o.RepoID),
+		LargeFiles: o.LargeFiles,
+		Symbols:    o.Symbols,
+		Branches:   branches,
+		Name:       o.Name,
+
+		Priority: o.Priority,
+
+		DocumentRanksVersion: o.DocumentRanksVersion,
+
+		Public:   o.Public,
+		Fork:     o.Fork,
+		Archived: o.Archived,
+
+		Error: o.Error,
+	}
+}
+
+func (s *sourcegraphClient) getIndexOptionsGRPC(fingerprint *proto.Fingerprint, repos []uint32) ([]indexOptionsItem, *proto.Fingerprint, error) {
```

Comment:
why does this not take a context?

Reply:
I'll tweak this to take a context, but this only used in the SourcegraphListResult.IterateIndexOptions callback (which doesn't take a context), so using the parent context (one of the parameters to List()) I don't think makes sense here.

Reply:
[sourcegraph/zoekt@`69d3558` (#551)](https://github.com/sourcegraph/zoekt/pull/551/commits/69d3558b3a7c91be6cdff93adea28c4d23f43c7d)